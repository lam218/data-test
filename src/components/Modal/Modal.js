//@flow

import React from "react";
import "./Modal.scss";

type Props = {
  doc: Object,
  closeModal: () => void,
};
const Modal = ({ doc, closeModal }: Props) => (
  <div className="modal">
    <div className="modal__outer">
      <div className="modal__inner">
        <div className="modal__button-container">
          <button onClick={closeModal} type="button">
            Close
          </button>
        </div>
        <div className="modal__content">
          <p>{doc.m_szDocBody}</p>
        </div>
        <div className="modal__button-container">
          <a className="modal__link" href={doc.m_szSrcUrl}>
            Read on original site
          </a>
        </div>
      </div>
    </div>
  </div>
);

export default Modal;
