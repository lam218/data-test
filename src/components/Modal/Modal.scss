@import "../../styles/placeholders.scss";

.modal {
  @extend %box-shadow;
  position: fixed;
  top: ($basepx * 5);
  z-index: 2;
  border-radius: $basepx;

  &__outer {
    margin: auto;
    width: calc(100% - 80px);
    background: $white;
    padding: ($basepx * 5);
  }
  &__inner {
    @extend %inner-container;
  }
  &__content {
    overflow-y: scroll;
    max-height: 70vh;
  }
  &__button-container{
      display: flex;
      justify-content: flex-end;
      padding: ($basepx * 5) 0;
  }
}
