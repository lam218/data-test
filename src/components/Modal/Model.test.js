import React from 'react';
import Modal from './Modal';
import renderer from 'react-test-renderer';


test('Test that the Modal renders', () => {
    const closeModal = jest.fn();
    const component = renderer.create(
        <Modal doc={{}} closeModal={closeModal}/>,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})