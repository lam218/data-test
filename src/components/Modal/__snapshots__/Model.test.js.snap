// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the Modal renders 1`] = `
<div
  className="modal"
>
  <div
    className="modal__outer"
  >
    <div
      className="modal__inner"
    >
      <div
        className="modal__button-container"
      >
        <button
          onClick={[MockFunction]}
          type="button"
        >
          Close
        </button>
      </div>
      <div
        className="modal__content"
      >
        <p />
      </div>
      <div
        className="modal__button-container"
      >
        <a
          className="modal__link"
        >
          Read on original site
        </a>
      </div>
    </div>
  </div>
</div>
`;
